src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,203 @@
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == 0) {
+            linkFirst(value);
+        } else if (index == size) {
+            add(value);
+        } else {
+            checkIndex(index);
+            linkBefore(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        if (index == 0) {
+            return unlinkFirst();
+        } else if (index == size - 1) {
+            return unlinkLast();
+        } else {
+            return unlink(index);
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        if (getIndex(object) != -1) {
+            remove(getIndex(object));
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private Node<T> prev;
+        private T item;
+        private Node<T> next;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Illegal index");
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        checkIndex(index);
+        Node<T> node;
+        if (index < (size >> 1)) {
+            node = first;
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+        } else {
+            node = last;
+            for (int i = size - 1; i > index; i--) {
+                node = node.prev;
+            }
+        }
+        return node;
+    }
+
+    private int getIndex(T object) {
+        int i = 0;
+        boolean isFound = false;
+        Node<T> temp = first;
+        for (; i < size; i++) {
+            if (object == null) {
+                if (temp.item == null) {
+                    isFound = true;
+                    break;
+                }
+                temp = temp.next;
+            } else {
+                if (object.equals(temp.item)) {
+                    isFound = true;
+                    break;
+                }
+                temp = temp.next;
+            }
+        }
+        return isFound ? i : -1;
```

Comment:
lets simplify it and return index when you find value, else just return -1 in you leave for-loop